### PR TITLE
account supports nested typed data sign

### PR DESCRIPTION
**Pull request by livingrockrises**

### CODE REVIEW

File: test/foundry/unit/concrete/erc1271/TestERC1271Account_IsValidSignature.t.sol [hunk 1]
```diff
@@ -107,6 +107,13 @@ contract TestERC1271Account_IsValidSignature is NexusTest_Base {
         ALICE_ACCOUNT.isValidSignature(originalHash, completeSignature);
     }
 
+    function testSupportsNestedTypedDataSign() public view {
```
Comment (Aboudjem) [inline test/foundry/unit/concrete/erc1271/TestERC1271Account_IsValidSignature.t.sol:110]:
```suggestion
    function test_SupportsNestedTypedDataSign() public view {
```

File: contracts/Nexus.sol [hunk 2]
```diff
@@ -342,6 +342,16 @@ contract Nexus is INexus, EIP712, BaseAccount, ExecutionHelper, ModuleManager, U
...
     }
 
+    /// @dev For automatic detection that the smart account supports the nested EIP-712 workflow.
+    /// By default, it returns `bytes32(bytes4(keccak256("supportsNestedTypedDataSign()")))`,
+    /// denoting support for the default behavior, as implemented in
+    /// `_erc1271IsValidSignatureViaNestedEIP712`, which is called in `isValidSignature`.
+    /// Future extensions should return a different non-zero `result` to denote different behavior.
+    /// This method intentionally returns bytes32 to allow freedom for future extensions.
+    function supportsNestedTypedDataSign() public view virtual returns (bytes32 result) {
+        result = bytes4(0xd620c85a);
```
Comment (Aboudjem) [inline contracts/Nexus.sol:352]: Can't we make it a pure function?

Comment (livingrockrises) [reply contracts/Nexus.sol:352]: done

**Comment by livingrockrises** [comment]
please approve

**Review by Aboudjem — APPROVED** [review]

**Pull request merged**